numpy/_core/src/multiarray/scalartypes.c.src

```diff
@@ -939,12 +939,23 @@ datetimetype_repr(PyObject *self)
         if (legacy_print_mode == -1) {
             return NULL;
         }
-        if (legacy_print_mode > 125) {
-            ret = PyUnicode_FromFormat("np.datetime64('%s')", iso);
-        }
-        else {
-            ret = PyUnicode_FromFormat("numpy.datetime64('%s')", iso);
+
+        PyObject *meta = metastr_to_unicode(&scal->obmeta, 1);
+        if((scal->obval == NPY_DATETIME_NAT) && (meta != NULL)){
+            if (legacy_print_mode > 125) {
```

Comment:
I guess technically we could define a new numpy 2.3 legacy print mode as well. I don't know if it's worth the ceremony just for this.